homeassistant/components/dobiss/light.py

```diff
@@ -0,0 +1,169 @@
+"""Support for dobiss lights."""
+
+import logging
+from typing import Any
+
+from dobissapi import DobissAnalogOutput, DobissLight, DobissOutput
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ENTITY_ID, ENTITY_MATCH_ALL, ENTITY_MATCH_NONE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import CONF_IGNORE_ZIGBEE_DEVICES, DOMAIN, KEY_API
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up dobiss lights."""
+    _LOGGER.debug("Setting up light component of %s", DOMAIN)
+
+    dobiss = hass.data[DOMAIN][config_entry.entry_id][KEY_API].api
+
+    light_entities = dobiss.get_devices_by_type(DobissLight)
```

Comment:
Why do you have to pass in a type? How many types are there? Would it make sense for the library to use an enum instead?